demos/init-db.php

```diff
@@ -20,8 +20,48 @@
 
 // a very basic file that sets up Agile Data to be used in some demonstrations
 
+trait ModelPreventModificationTrait
+{
+    public function atomic(\Closure $fx)
+    {
+        $eRollback = new \Exception('Prevent modification');
+        $res = null;
+        try {
+            parent::atomic(function () use ($fx, $eRollback, &$res) {
+                $res = $fx();
+
+                throw $eRollback;
```

Comment:
FYI this guarantees all save (insert/update/delete) operations will be executed, but always rolled back